src/store/connectedHotspot/connectedHotspotSlice.ts

```diff
@@ -87,15 +101,33 @@ export const fetchHotspotActivity = createAsyncThunk<
   return list.takeJSON(opts.fetchCount || 10)
 })
 
-export const fetchHotspotDetails = createAsyncThunk<Hotspot, HotspotDetails>(
-  'account/fetchHotspotDetails',
-  async (details, thunkAPI) => {
-    thunkAPI.dispatch(
-      connectedHotspotSlice.actions.initConnectedHotspot(details),
-    )
-    return getHotspotDetails(details.address || '')
-  },
-)
+export type AllHotspotDetails = {
+  hotspot: Hotspot
+  onboardingRecord: OnboardingRecord
+}
+export const fetchHotspotDetails = createAsyncThunk<
+  AllHotspotDetails,
+  HotspotDetails
+>('account/fetchHotspotDetails', async (details, thunkAPI) => {
+  thunkAPI.dispatch(connectedHotspotSlice.actions.initConnectedHotspot(details))
+
+  if (!details.address) {
+    throw new Error('fetchHotspotDetails address is missing')
+  }
+  if (!details.onboardingAddress) {
+    throw new Error('fetchHotspotDetails onboardingAddress is missing')
+  }
+
+  const vals = await Promise.all([
```

Comment:
you could potentially destructure this array:

```
const [hotspot, onboardingRecord] = await Promise.all //...
```

saves a const declaration and is more descriptive than vals